codehost/github/target/issue_target.go

```
@@ -39,7 +39,13 @@ func (t *IssueTarget) GetNodeID() string {
 	return t.issue.GetNodeID()
 }
 
-func (t *IssueTarget) Close() error {
+func (t *IssueTarget) Close(comment string) error {
+	if comment != "" {
```

I think it makes more sense for this validation to happen the `t.Comment` function.

---

It would actually be great to document in the Comment function what happens if the comment the empty string.

---

Idk if that would be desirable because whenever we comment `""` this action would be ignored, following your suggestion.
But, for now, we only want this to happen in the particular case of `Close()` in other cases we may want to throw an error on  other comments like `""` like the already defined default behavior. For example, `$comment("")` should return an error, should not?

---

I think we need to check what exactly happens if a user tries $comment(""). I don't think it should return an error in that case.

---

I took the decision of implementing that way because github throws an error saying that a comment can't have an empty body

---

@marcelosousa I tried executing `$comment("")` in the PR https://github.com/explore-dev/didactic-robot/pull/68 and as expected a fail happens.
Log message: `Error: error running reviewpad team edition. Details POST https://api.github.com/repos/explore-dev/didactic-robot/issues/68/comments: 422 Validation Failed [{Resource:IssueComment Field:data Code:unprocessable Message:Body cannot be blank}]`

---

I still believe `t.Comment()` should remain unchanged since this function has a direct purpose: create a comment.
If it doesn't have the conditions to do so (e.g., empty comment) it should fail - a comment CANNOT have a blank comment or else it fails to be a comment.
Said that, the validation you mentioned on the first comment of this thread, only makes sense to be made in `t.Close()` since this function can have two behaviours:
1. Perform a close WITH comment
2. Perform a close WITHOUT comment
It is a particular case.